server/middlewares/setCookies.js

```diff
@@ -0,0 +1,13 @@
+const { sign } = require('jsonwebtoken');
+
+module.exports = (req, res, next) => {
+  const { userId } = req;
+  const { secretKey } = process.env;
+  sign({ role: 'user', userId }, secretKey, (err, token) => {
+    if (err) {
+      next(err);
+    } else {
+      res.cookie('access_token', token, { httpOnly: true, secure: true }).send();
```

Comment:
Here we send a cookie then we finish the response, why? should we use next() instead? then send a message successfully signed in then in the front we redirect the user to the cart page